### PR TITLE
Fix closing cursor with no previously running query

### PR DESCRIPTION
**Pull request by hovaesco**
## Description

Fixes https://github.com/trinodb/trino-python-client/issues/428

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: tests/integration/test_dbapi_integration.py [hunk 1]
```diff
@@ -1343,23 +1343,13 @@ def test_cancel_query(trino_connection):
     cur.fetchone()
     cur.cancel()  # would raise an exception if cancel fails
 
-    cur = trino_connection.cursor()
-    with pytest.raises(Exception) as cancel_error:
-        cur.cancel()
-    assert "Cancel query failed; no running query" in str(cancel_error.value)
```
Comment (hashhar) [inline tests/integration/test_dbapi_integration.py:1349]: can you adjust the test to instead make sure this next cancel doesn't fail with exception?

Comment (hashhar) [reply tests/integration/test_dbapi_integration.py:1349]: same below

**Review by hashhar — APPROVED** [review]
LGTM % tests

File: tests/integration/test_dbapi_integration.py (hunk 1 above)
Comment (hovaesco) [reply tests/integration/test_dbapi_integration.py]: done

**Pull request merged**